etc/tests/test_bootstrapper.py

```diff
@@ -163,6 +164,183 @@ def test_main_method_with_dir_outputs(monkeypatch, s3_setup, tmpdir):
     main_method_setup_execution(monkeypatch, s3_setup, tmpdir, argument_dict)
 
 
+def test_process_metrics_method_no_metadata_file(monkeypatch, s3_setup, tmpdir):
+    """Test for process_metrics
+
+    Verifies that process_metrics produces a valid KFP UI metadata file if
+    the node's script | notebook did not generate this metadata file.
+    """
+    argument_dict = {'cos-endpoint': 'http://' + MINIO_HOST_PORT,
+                     'cos-bucket': 'test-bucket',
+                     'cos-directory': 'test-directory',
+                     'cos-dependencies-archive': 'test-archive.tgz',
+                     'filepath': 'etc/tests/resources/test-notebookA.ipynb',
+                     'inputs': 'test-file.txt;test,file.txt',
+                     'outputs': 'test-file/test-file-copy.txt;test-file/test,file/test,file-copy.txt',
+                     'user-volume-path': None}
+
+    metadata_file = '/tmp/mlpipeline-ui-metadata.json'
```

Comment:
Same as previous, although you probably want `os.path.join(tempfile.gettempdir(), 'mlpipeline-ui-metadata.json')` here.

Reply:
The test will now be skipped if the environment doesn't meet the file system prerequisites.